source/inputCore.py

```diff
@@ -526,7 +526,14 @@ def executeGesture(self, gesture):
 		immediate = getattr(gesture, "_immediate", True)
 		speechEffect = gesture.speechEffectWhenExecuted
 		if speechEffect == gesture.SPEECHEFFECT_CANCEL:
-			queueHandler.queueFunction(queueHandler.eventQueue, speech.cancelSpeech, _immediate=immediate)
+			# Import late to avoid circular import.
+			import braille
+			with braille.handler.suppressClearBrailleRegions(script):
+				queueHandler.queueFunction(
+					queueHandler.eventQueue,
+					speech.cancelSpeech,
+					_immediate=immediate,
+				)

```

Comment:
I don't believe this has the intended affect you are going for here.

The context manager will only be in use when queuing the function, not while the function executes.

Instead, `speech.cancelSpeech` should be wrapped with the context manager.

```suggestion
			@braille.handler.suppressClearBrailleRegions(script):
			def suppressCancelSpeech():
				speech.cancelSpeech()

			queueHandler.queueFunction(
				queueHandler.eventQueue,
				suppressCancelSpeech,
				_immediate=immediate,
			)
```

Reply:
>I tested it, and it works as it should, as far as I know.

@Emil-18 - could you please explain what you tested here?

To confirm this works as intended, could you re-test with the code modified as follows?
This simulates slow/late processing of cancel speech.


```suggestion
			def delayedCanceledSpeech():
				import time
				time.sleep(3)
				speech.cancelSpeech()

			with braille.handler.suppressClearBrailleRegions(script):
				queueHandler.queueFunction(
					queueHandler.eventQueue,
					delayedCanceledSpeech,
					_immediate=False,
				)
```

Reply:
@Emil-18 - can you please respond to this comment

Reply:
If you don't want to test this, please accept the original code suggestion

Reply:
This still has not been committed - please do not mark this as resolved.

Reply:
Thanks I see this has been committed now